### PR TITLE
Add `webcodecs` IDL spec and generated code

### PR DESCRIPTION
**Pull request by MaxDesiatov**
This is a requirement for `webgl2`, which I'm going to add in a future PR.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@j-f1 I needed a manual change in 6781702 to fix the absence of `BitrateMode`, not sure if this a bug in IDL, or the parser, or our generator?

**Comment by j-f1** [comment]
I recommend temporarily changing `parse-all.js` to output `JSON.stringify(parsedFiles)`, and then running `node parse-all.js > specs.json`. You’ll be able to search for `"name": "BitrateMode"` in the resulting file, and then use VS Code’s “Go to Parent Fold” command to find that the `BitrateMode` type is defined in the `mediastream-recording` spec.

**Comment by MaxDesiatov** [comment]
Marking this as draft until https://github.com/swiftwasm/DOMKit/pull/19 is resolved.

File: Sources/DOMKit/Generated.swift [hunk 1]
```diff
@@ -590,682 +608,611 @@ public class Attr: Node {
     public var specified: Bool
 }
 
-public class AudioTrack: JSBridgedClass {
-    @inlinable public class var constructor: JSFunction { JSObject.global[Strings.AudioTrack].function! }
+public class AudioData: JSBridgedClass {
+    @inlinable public class var constructor: JSFunction { JSObject.global[Strings.AudioData].function! }
```
Comment (j-f1) [inline Sources/DOMKit/Generated.swift:612]: It looks like the types are being printed in different orders each time. This doesn’t have to be fixed for this PR tho, I’ll try to open a PR shortly

**Review by j-f1 — APPROVED** [review]

File: Sources/DOMKit/Generated.swift (hunk 1 above)
Comment (j-f1) [reply Sources/DOMKit/Generated.swift:612]: looking at this again (especially after reviewing the other PRs), I don’t think this is an actual problem. Not sure why the diff got messed up here but all the other PRs solely contain additive changes.

**Comment by j-f1** [comment]
Heads up that a few types have been disabled because they depend on specs not included in the initial PR, so feel free to remove any of these that work now:

https://github.com/swiftwasm/WebAPIKit/blob/45f476a7ca582d8ebf5623c4493aa4194b416bb2/Sources/WebIDLToSwift/IDLBuilder.swift#L19-L22

https://github.com/swiftwasm/WebAPIKit/blob/45f476a7ca582d8ebf5623c4493aa4194b416bb2/Sources/WebIDLToSwift/IDLBuilder.swift#L96-L99

**Comment by MaxDesiatov** [comment]
Checked that list, only `CanvasImageSource` can be auto-generated cleanly. `OffscreenRenderingContext` needs full WebGL2 spec to work in a related enum.

**Pull request merged**